### PR TITLE
feat: Improved compliance of URL object

### PR DESCRIPTION
**Pull request by nabetti1720**
### Description of changes

Add the following methods to improve the compliance of `URL` Object.

- URL.toJSON()

### Checklist

- [x] Created unit tests in `tests/unit` and/or in Rust for my feature if needed
- [x] Ran `make fix` to format JS and apply Clippy auto fixes
- [x] Made sure my code didn't add any additional warnings: `make check`
- [x] Updated documentation if needed ([API.md](API.md)/[README.md](README.md)/Other)

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: llrt_core/src/modules/http/url.rs [hunk 1]
```diff
@@ -238,6 +238,13 @@ impl<'js> URL<'js> {
         self.password.clone_from(&password);
         password
     }
+
+    #[qjs(rename = PredefinedAtom::ToJSON)]
+    fn to_json(&self) -> String {
+        // NOTE: It is not wrong to return as a string.
```
Comment (richarddavison) [inline llrt_core/src/modules/http/url.rs:244]: I think we can remove this comment!

**Review by richarddavison — CHANGES_REQUESTED** [review]
Thanks, LGTM! Just remove that comment and we're good! 🎉

File: llrt_core/src/modules/http/url.rs (hunk 1 above)
Comment (nabetti1720) [reply llrt_core/src/modules/http/url.rs:244]: Do you want me to delete the next comment as well?
```rust
// https://developer.mozilla.org/en-US/docs/Web/API/URL/toJSON
```

Comment (richarddavison) [reply llrt_core/src/modules/http/url.rs]: Thats fine to keep!

**Review by richarddavison — APPROVED** [review]

**Pull request merged**